src/index.html

```diff
@@ -1,43 +1,174 @@
-<html>
+<!DOCTYPE html>
+<html lang="$language_short">
 <head>
-    <base href="/">
-    <link rel="stylesheet" href="/style.css">
+	<meta charset="UTF-8" />
+	<meta name="viewport" content="width=device-width, initial-scale=1" />
+	<title>$page_title</title>
+	<meta name="description" content="$page_description" />
+	<link rel="canonical" href="$page_url" />
+	<meta name="og:title" content="$page_title" />
+	<meta name="og:description" content="$page_description" />
+	<meta name="og:url" content="$page_url" />
+	<meta name="og:image" content="$og_image" />
+	<meta name="og:image:alt" content="$og_alt" />
+	<meta name="article:published_time" content="$page_date" />
+	<meta name="article:modified_time" content="$page_date" />
+	<link rel="preload" href="$optional_webfont" as="font" type="font/woff2" crossorigin />
+	<meta name="robots" content="index, follow, max-image-preview:large, max-snippet:-1, max-video-preview:-1" />
+	<meta name="og:site_name" content="$site_name" />
+	<meta name="og:type" content="website" />
+	<meta name="og:locale" content="$language_long" />
+	<meta name="og:image:width" content="1200" />
+	<meta name="og:image:height" content="630" />
+	<meta name="og:email" content="$site_email" />
+	<meta name="og:phone_number" content="$site_telephone" />
+	<meta name="article:publisher" content="$site_name" />
+	<meta name="fb:app_id" content="$facebook_appid" >
+	<meta name="twitter:site" content="$twitter_site">
+	<meta name="twitter:creator" content="$twitter_account">
+	<meta name="twitter:card" content="summary_large_image" />
+	<link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png" />
+	<link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png" />
```

Comment:
if this is intended to be used for portofolios or smll js demobs, not sure we should include this many branding images, the user will need to change/remove all of them, simplicity should be a focus of the project

Reply:
Unfortunately, when it comes to standardizing the meta images and icons, this is necessary to cover all the browsers and versions. I will link to https://realfavicongenerator.net/ so it's as easy as possible to generate everything based on a single image.